libtiledbsoma/src/utils/arrow_adapter.cc

```diff
@@ -1091,68 +1095,98 @@ ArraySchema ArrowAdapter::tiledb_schema_from_arrow_schema(
-                        platform_config);
-
-                    // Do not set the `isattr` flag to false to add the
-                    // `soma_geometry` as attribute
+                        child,
+                        spatial_column_info.second.get(),
+                        spatial_column_info.first.get(),
+                        soma_type,
+                        type_metadata,
+                        platform_config));
                 } else {
-                    auto dim = tiledb_dimension_from_arrow_schema(
+                    columns.push_back(SOMADimension::create(
                         ctx,
                         index_column_schema->children[i],
                         index_column_array->children[i],
                         soma_type,
                         type_metadata,
-                        "",
-                        "",
-                        platform_config);
-                    dims.insert({dim.name(), dim});
-                    isattr = false;
+                        platform_config));
                 }
-
+                isattr = false;
+                LOG_DEBUG(std::format(
+                    "[ArrowAdapter] adding dimension {}", child->name));
                 break;
             }
         }
 
         if (isattr) {
-            auto attr = tiledb_attribute_from_arrow_schema(
-                ctx, child, type_metadata, platform_config);
-            if (attr.second.has_value()) {
-                ArraySchemaExperimental::add_enumeration(
-                    *ctx, schema, attr.second.value());
-            }
-
+            columns.push_back(SOMAAttribute::create(
+                ctx, child, type_metadata, platform_config));
             LOG_DEBUG(
                 std::format("[ArrowAdapter] adding attribute {}", child->name));
-
-            schema.add_attribute(attr.first);
         }
     }
 
+    // Unit tests expect dimension order should match the index column schema
+    // and NOT the arrow schema
     for (int64_t i = 0; i < index_column_schema->n_children; ++i) {
         LOG_DEBUG(std::format("[ArrowAdapter] child {}", i));
-        auto col_name = index_column_schema->children[i]->name;
-        if (strcmp(col_name, SOMA_GEOMETRY_COLUMN_NAME.c_str()) == 0) {
-            for (auto& dim : dims) {
-                if (dim.first.substr(std::max(0ul, dim.first.size() - 5)) ==
-                    std::string("__min")) {
-                    domain.add_dimension(dim.second);
-                }
+        auto column = std::find_if(
+            columns.begin(), columns.end(), [&](auto col) {
+                return strcmp(
+                           col->name().c_str(),
+                           index_column_schema->children[i]->name) == 0;
+            });
+
+        if (column == columns.end()) {
+            throw TileDBSOMAError(std::format(
+                "[ArrowAdapter][tiledb_schema_from_arrow_schema] Index column "
+                "{} missing",
```

Comment:
```suggestion
                "'{}' missing",
```

Strings in error messages should always be quoted/bracketed. Even if you think it's impossible for the string to ever be empty. Heaven forbid someday there is some bug somewhere somehow ... and an empty string gets in here ... that needs to be clear to everyone that sees the error message.